ragna/core/_rag.py

```diff
@@ -59,35 +69,110 @@ def _load_component(self, component: Union[Type[C], C]) -> C:
 
         return self._components[cls]
 
-    def chat(
+    def corpus(
         self,
         *,
         documents: Iterable[Any],
         source_storage: Union[Type[SourceStorage], SourceStorage],
+        corpus_name: str,
+        **params: Any,
+    ) -> Corpus:
+        """Create a new [ragna.core.Corpus][].
+
+        Args:
+            documents: Documents to use.
+            source_storage: Source storage to use.
+            corpus_name: Name of the corpus.
+            **params: Additional parameters passed to the source storage.
+        """
+        return Corpus(
+            self,
+            documents=documents,
+            source_storage=source_storage,
+            corpus_name=corpus_name,
+            **params,
+        )
+
+    def chat(
+        self,
+        *,
+        corpus: Corpus,
         assistant: Union[Type[Assistant], Assistant],
         **params: Any,
     ) -> Chat:
         """Create a new [ragna.core.Chat][].
 
         Args:
-            documents: Documents to use. If any item is not a [ragna.core.Document][],
-                [ragna.core.LocalDocument.from_path][] is invoked on it.
-            source_storage: Source storage to use.
+            corpus: Corpus to use.
             assistant: Assistant to use.
-            **params: Additional parameters passed to the source storage and assistant.
+            **params: Additional parameters passed to the assistant.
         """
         return Chat(
             self,
-            documents=documents,
-            source_storage=source_storage,
+            corpus=corpus,
             assistant=assistant,
             **params,
         )
 
 
+class Corpus:
+    """
+    !!! tip
+
+        This object is usually not instantiated manually, but rather through
+        [ragna.core.Rag.corpus][].
+
+        Args:
+        rag: The RAG workflow this corpus is associated with.
+        documents: Documents to use.
+        source_storage: Source storage to use.
+        **params: Additional parameters passed to the source storage.
+    """
+
+    def __init__(
+        self,
+        rag: Rag,
+        *,
+        documents: Iterable[Any],
+        source_storage: Union[Type[SourceStorage], SourceStorage],
+        corpus_name: str,
```

Comment:
At least in the "user just passes document paths", this needs to have a default.